dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import static java.util.stream.Collectors.toList;
+
+import com.palantir.dialogue.Channel;
+import java.util.Collection;
+import java.util.List;
+
+public final class Channels {
+
+    private Channels() {}
+
+    public static Channel create(Collection<? extends Channel> channels) {
```

Comment:
let's make this package-private

Reply:
I can't do that right now, because the constructors that people should really use are in different projects. eg: if you want to create an HttpClient backed Channel then you need to use the constructor in the `dialogue-java-client` project, which in turn will call this. This is only the case right now because we're supporting both the Java and OkHttp clients.